### ISSUE TITLE
optionalDependencies handled differently for non-registry sources

### ISSUE THREAD

**Issue by clefru** [svanderburg/node2nix#137]
From the [npm manual on optionalDependencies](https://npm.github.io/using-pkgs-docs/package-json/types/optionaldependencies.html):
```
Entries in optionalDependencies will override entries of the same name 
in dependencies, so it's usually best to only put in one place.
```
spark-wallet's [package.json](https://github.com/shesek/spark-wallet/blob/master/package.json#L59) follows that. It lists qrcode-terminal in `optionalDependencies` but not in `dependencies`.

Both of those commands install qrcode-terminal
```
npm install https://registry.npmjs.org/spark-wallet/-/spark-wallet-0.2.5.tgz # tgz-driven install
npm install spark-wallet@0.2.5 # registry-driven install
```
For node2nix, the behavior is different. In this [commit](https://github.com/jonasnick/nix-bitcoin/commit/2250644f829d405c8cb4315959b86b8d0c6d08fd#diff-eeda76e2ba3156480c627b424f9367b9) I have to swap out a tgz driven install for a registry driven install. The reason is with tgz-driven installs node2nix consults the tgz package.json which does NOT contain qrcode-terminal. Hence it fails to install it. For the registry-driven install, node2nix queries the registry. The npmjs registry normalizes the package.json data in that `optionalDependencies` are also listed in `dependencies`, despite the original package.json not doing that. Grab for `dependencies` in [here](http://registry.npmjs.org/spark-wallet) to see that.

It would be good if node2nix would mimic that for non-registry installs.

**Comment by svanderburg** [comment]
Yes that indeed would be nice, but not very easy to replicate. Currently, node2nix ignores optionalDependencies altogether -- it simply uses whatever is in the dependencies section of package.json file or metadata that comes from the registry.

optionalDependecy deployments are also allowed to fail (deterministically and non-deterministically). That's also something we need to deal with.

**Comment by nixbitcoin** [comment]
Could one use supplements to pass the missing optionalDependencies? I tried adding my missing dependency to `supplement.json` without changing the `default.nix` or creating an `override.nix` and kept on getting `Error: Cannot find module`
